test/stdlib_dependencies.jl

```diff
@@ -1,148 +1,222 @@
-using Pkg, Test, Libdl
+using Libdl
+using Pkg
+using Test
+prev_env = Base.active_project()
+Pkg.activate(temp=true)
+Pkg.add(Pkg.PackageSpec(name="ObjectFile", uuid="d8793406-e978-5875-9003-1fc021f44a92"))
+using ObjectFile
+try
 
-# Remove `.X.dylib` or just `.dylib`
-function strip_soversion_macos(lib)
-    m = match(r"^(.*?)(\.\d+)*\.dylib$", lib)
-    if m !== nothing
-        return m.captures[1]
+    # Remove `.X.dylib` or just `.dylib`
+    function strip_soversion_macos(lib)
+        m = match(r"^(.*?)(\.\d+)*\.dylib$", lib)
+        if m !== nothing
+            return m.captures[1]
+        end
+        return lib
     end
-    return lib
-end
 
-# Remove `.so.X` or just `.so`
-function strip_soversion_linux(lib)
-    m = match(r"^(.*?)\.so(\.\d+)*$", lib)
-    if m !== nothing
-        return m.captures[1]
+    # Remove `.so.X` or just `.so`
+    function strip_soversion_linux(lib)
+        m = match(r"^(.*?)\.so(\.\d+)*$", lib)
+        if m !== nothing
+            return m.captures[1]
+        end
+        return lib
     end
-    return lib
-end
 
-# Remove `-X.dll` or just `.dll`
-function strip_soversion_windows(lib)
-    m = match(r"^(.*?)(-\d+)*\.dll$", lib)
-    if m !== nothing
-        return m.captures[1]
+    # Remove `-X.dll` or just `.dll`
+    function strip_soversion_windows(lib)
+        m = match(r"^(.*?)(-\d+)*\.dll$", lib)
+        if m !== nothing
+            return m.captures[1]
+        end
+        return lib
     end
-    return lib
-end
```

Comment:
You can use `Base.BinaryPlatforms.parse_dl_name_version` instead of all of this?